src/main/java/io/github/moulberry/notenoughupdates/auction/APIManager.java

```diff
@@ -409,6 +409,7 @@ private void updatePageTickShort() {
 				int page = pagesToDownload.pop();
 				getPageFromAPI(page);
 			} catch (NoSuchElementException ignored) {
+				return;
```

Comment:
ideally we would synchronize the entire list in another thread, but ig this is fine